### ISSUE TITLE
kontena master logs

### ISSUE THREAD

**Issue by matti** [kontena/kontena#1842]
Some stuff is just not available in grid/stack/service logs

**Comment by colinmollenhour** [comment]
I agree, it would be very handy to be able to fetch these logs from the kontena CLI instead of having to SSH into the master. Also and more importantly for security reasons I'd like to make the logs accessible to more users (so they can troubleshoot issues) but without having to give them all direct SSH access to the master.

**Comment by SpComb** [comment]
Ideally errors and log messages within the server should be associated with specific grids/stacks/services, and then those messages could be included in the API responses for things like stack/service deployments and shown directly by the CLI.

It might also make some sense to have the server buffer the `Logging` messages in memory, and provide those via some API...

**Comment by SpComb** [comment]
The current SSH-based method for this BTW is `kontena master ssh docker logs kontena-server-api`.

**Comment by SpComb** [comment]
Kontena 1.2 will have some level of server+agent logging via #2028  `kontena grid/service events`
